src/TTrigger.cpp

```diff
@@ -837,38 +855,42 @@ bool TTrigger::match(char* subject, const QString& toMatch, int line, int posOff
         }
 
         int size = mRegexCodePropertyList.size();
-        for (int i = 0;; i++) {
-            if (i >= size) {
+        for (int patternNumber = 0;; patternNumber++) {
```

Comment:
This was a bit offtopic but poorly named iterators make life unnecessarily difficult.

Reply:
Wanna do the same in `void dlgTriggerEditor::slot_trigger_selected(QTreeWidgetItem* pItem)`? 😉

Reply:
Nop, keeping offtopic to a minimum. It should be a separate PR to go through and improve it, otherwise the PR's scope blows up and what gets merged in the end won't just be a 'add a prompt trigger pattern type' but 'add a prompt trigger pattern type, and improve a, fixup b, cleanup d' all at once - which as you see is messy!

Reply:
I guess so... 🙂